File: brokenswords.py
import math

n = int(input())

tb = 0
lr = 0

for x in range(n):
    s = input()
    if s[0] == "0":
        tb += 1
    if s[1] == "0":
        tb += 1
    if s[2] == "0":
        lr += 1
    if s[3] == "0":
        lr += 1

t = math.floor(min(tb, lr) / 2)
tb = tb - t * 2
lr = lr - t * 2
print(t)
print(tb)
print(lr)

File: brokenswords.cpp
#include <iostream>
#include <math.h> 
using namespace std;

int main() {
    int n;
    cin >> n;

    int tb = 0;
    int lr = 0;

    for (int i=0; i < n; i++) {
        string s;
        cin >> s;
        if (s.at(0) == '0') {
            tb++;
        }
        if (s.at(1) == '0') {
            tb++;
        }
        if (s.at(2) == '0') {
            lr++;
        }
        if (s.at(3) == '0') {
            lr++;
        }
    }
    int t = floor(min(tb, lr) / 2);
    tb = tb - t * 2;
    lr = lr - t * 2;
    cout << t << " " << tb << " " << lr << endl;
    return 0;
}